File: lightrag_webui/api/health.js
export default function handler(req, res) {
  // Set CORS headers
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, POST, PUT, DELETE, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type, Authorization, X-API-Key');

  if (req.method === 'OPTIONS') {
    res.status(200).end();
    return;
  }

  if (req.method === 'GET') {
    res.status(200).json({
      status: 'healthy',
      working_directory: './rag_storage',
      input_directory: './inputs',
      configuration: {
        llm_binding: 'openai',
        llm_model: 'gpt-4o-mini',
        embedding_binding: 'openai',
        embedding_model: 'text-embedding-3-large',
        max_tokens: 16384,
        kv_storage: 'JsonKVStorage',
        doc_status_storage: 'JsonDocStatusStorage',
        graph_storage: 'NetworkXStorage',
        vector_storage: 'NanoVectorDBStorage'
      },
      pipeline_busy: false,
      webui_title: 'Augentik Dashboard',
      webui_description: 'Audit Intelligence Platform'
    });
  } else {
    res.status(405).json({ error: 'Method not allowed' });
  }
}

File: lightrag_webui/api/health.py
from fastapi import FastAPI
from fastapi.responses import JSONResponse
import os

app = FastAPI()

@app.get("/api/health")
async def health_check():
    return JSONResponse({
        "status": "healthy",
        "message": "Augentik API is running",
        "version": "1.0.0"
    })

@app.get("/api/auth-status")
async def auth_status():
    return JSONResponse({
        "auth_configured": False,
        "access_token": "demo-token",
        "auth_mode": "disabled",
        "message": "Demo mode - no authentication required",
        "webui_title": "Augentik Dashboard",
        "webui_description": "Audit Intelligence Platform"
    })

# Vercel serverless function handler
def handler(request):
    return app(request)
